pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -214,61 +214,66 @@ func (admitter *VMsAdmitter) applyInstancetypeToVm(vm *v1.VirtualMachine) []meta
 func (admitter *VMsAdmitter) authorizeVirtualMachineSpec(ar *admissionv1.AdmissionRequest, vm *v1.VirtualMachine) ([]metav1.StatusCause, error) {
 	var causes []metav1.StatusCause
 
-	for idx, dataVolume := range vm.Spec.DataVolumeTemplates {
-		cloneSource, err := typesutil.GetCloneSourceWithInformers(vm, &dataVolume.Spec, admitter.DataSourceInformer)
-		if err != nil {
-			causes = append(causes, metav1.StatusCause{
-				Type:    metav1.CauseTypeUnexpectedServerResponse,
-				Message: err.Error(),
-				Field:   k8sfield.NewPath("spec", "dataVolumeTemplates").Index(idx).String(),
-			})
+	// Skip DataVolumeTemplates validation if we want to delete the vm
+	// This allows the vm controller to remove the finalizers
+	// also when, for example, the datasource is already gone.
+	if vm.DeletionTimestamp == nil || len(vm.Finalizers) > 0 {
```

Comment:
Is it enough to have the first part of the condition?

Reply:
Even better with an early return :)
```
if vm.DeletionTimestamp != nil {
	return causes, nil
}
```
Thanks